src/Angor/Client/Pages/Spend.razor

```diff
@@ -766,89 +766,104 @@
         }
     }
 
-    Dictionary<string, Transaction> trxsHexCache = new();
-
+    private bool updateUnconfirmedOutbound = false;
     private async Task CheckSpentFund()
     {
-        List<QueryTransaction> trxs = new();
         var unconfirmedOutbound = _cacheStorage.GetUnconfirmedOutboundFunds();
         bool updateUnconfirmedOutbound = false;
-
+        List<Task> tasks = [];
         foreach (StageData stageData in stageDatas)
         {
             foreach (var item in stageData.Items)
             {
-                // if the item is already spent by founder or investor, we skip it
-                // because we have already processed this item and we know the outcome.
-                if (item.IsSpent && ((item.SpentType != "founder") || (item.SpentType != "investor")))
-                    continue;
+                tasks.Add(processStageItem(item, stageData, unconfirmedOutbound));
+            }
+        }
+        await Task.WhenAll(tasks);
+        if (updateUnconfirmedOutbound)
+        {
+            _cacheStorage.SetUnconfirmedOutboundFunds(unconfirmedOutbound);
+        }
+    }
+    private Dictionary<string, Task<QueryTransaction>> queryTransactionCache = [];
+    private Dictionary<string, Task<Transaction>> transactionCache = [];
+
+    private async Task<QueryTransaction> GetQueryTransactionCache(string txid)
+    {
+        if (queryTransactionCache.TryGetValue(txid, out var cachedTask))
+            return await cachedTask;
 
-                QueryTransaction? trx = trxs.FirstOrDefault(f => f.TransactionId == item.Trxid);
+        var task = _IndexerService.GetTransactionInfoByIdAsync(txid);
+        queryTransactionCache[txid] = task;
+        return await task;
+    }
 
-                if (trx == null)
-                {
-                    trx = await _IndexerService.GetTransactionInfoByIdAsync(item.Trxid);
-                    trxs.Add(trx);
-                }
+    private async Task<Transaction> GetTransactionCache(string txid)
+    {
+        if (transactionCache.TryGetValue(txid, out var cachedTask))
+            return await cachedTask;
 
-                var output = trx.Outputs.First(outp => outp.Index == item.Outputindex);
+        var task = Task.Run(async () =>
+        {
+            var fetchTrx = await _IndexerService.GetTransactionHexByIdAsync(txid);
```

Comment:
Better to only return the task from the indexer service and call the network create transaction somewhere else